x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CrossClusterEsqlRCS1UnavailableRemotesIT.java

```diff
@@ -0,0 +1,283 @@
+            .build();
+    }
+
+    @ClassRule
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    @Before
+    public void setupPreRequisites() throws IOException {
+        setupRolesAndPrivileges();
+        loadData();
+    }
+
+    public void testEsqlRcs1UnavailableRemoteScenarios() throws Exception {
+        clusterShutDownWithRandomSkipUnavailable();
+        remoteClusterShutdownWithSkipUnavailableTrue();
+        remoteClusterShutdownWithSkipUnavailableFalse();
+    }
+
+    private void clusterShutDownWithRandomSkipUnavailable() throws Exception {
+        // skip_unavailable is set to a random boolean value.
+        // However, no clusters are closed. Hence, we do not expect any other behaviour
+        // other than a 200-OK.
+
+        configureRemoteCluster("my_remote_cluster", fulfillingCluster, true, randomBoolean(), randomBoolean());
+        String query = "FROM *,my_remote_cluster:* | LIMIT 10";
+        Response response = client().performRequest(esqlRequest(query));
+
+        Map<String, Object> map = responseAsMap(response);
+        ArrayList<?> columns = (ArrayList<?>) map.get("columns");
+        ArrayList<?> values = (ArrayList<?>) map.get("values");
+        Map<?, ?> clusters = (Map<?, ?>) map.get("_clusters");
+        Map<?, ?> clusterDetails = (Map<?, ?>) clusters.get("details");
+        Map<?, ?> localClusterDetails = (Map<?, ?>) clusterDetails.get("(local)");
+        Map<?, ?> remoteClusterDetails = (Map<?, ?>) clusterDetails.get("my_remote_cluster");
+
+        assertOK(response);
+        assertThat((int) map.get("took"), greaterThan(0));
+        assertThat(columns.size(), is(4));
+        assertThat(values.size(), is(9));
+
+        assertThat((int) clusters.get("total"), is(2));
+        assertThat((int) clusters.get("successful"), is(2));
+        assertThat((int) clusters.get("running"), is(0));
+        assertThat((int) clusters.get("skipped"), is(0));
+        assertThat((int) clusters.get("partial"), is(0));
+        assertThat((int) clusters.get("failed"), is(0));
+
+        assertThat(clusterDetails.size(), is(2));
+        assertThat((int) localClusterDetails.get("took"), greaterThan(0));
+        assertThat(localClusterDetails.get("status"), is("successful"));
+
+        assertThat((int) remoteClusterDetails.get("took"), greaterThan(0));
+        assertThat(remoteClusterDetails.get("status"), is("successful"));
+    }
+
+    private void remoteClusterShutdownWithSkipUnavailableTrue() throws Exception {
+        // Remote cluster is closed and skip unavailable is set to true.
+        // We expect no exception and partial results from the remaining open cluster.
+
+        configureRemoteCluster("my_remote_cluster", fulfillingCluster, true, randomBoolean(), true);
+
+        try {
+            // Close remote cluster.
+            fulfillingCluster.stop(true);
+
+            // A random query that targets our remote cluster.
```

Comment:
nit: I wouldn't say "random query" here since you don't have any randomization it. You can probably just delete the comment or just say "a simple query that includes the remote cluster".